python/tvm/meta_schedule/builder/local_builder.py

```diff
@@ -0,0 +1,224 @@
+    timeout_sec: float
+    f_build: Optional[str]
+    f_export: Optional[str]
+
+    def __init__(
+        self,
+        *,
+        max_workers: Optional[int] = None,
+        timeout_sec: float = 30.0,
+        f_build: str = None,
+        f_export: str = None,
+        initializer: Optional[Callable[[], None]] = None,
+    ) -> None:
+        """Constructor.
+
+        Parameters
+        ----------
+        max_workers : Optional[int]
+            The maximum number of worker processes to be used.
+            Defaults to number of CPUs.
+        timeout_sec : float
+            The timeout in seconds for the build.
+        f_build : Optional[str]
+            Name of the build function to be used.
+            Defaults to `meta_schedule.builder.default_build`.
+            The signature is Callable[[IRModule, Target], Module].
+        f_export : Optional[str]
+            Name of the export function to be used.
+            Defaults to `meta_schedule.builder.default_export`.
+            The signature is Callable[[Module], str].
+        initializer : Optional[Callable[[], None]]
+            The initializer to be used for the worker processes.
+        """
+        super().__init__()
+
+        if max_workers is None:
+            max_workers = cpu_count()
+
+        self.pool = PopenPoolExecutor(
+            max_workers=max_workers,
+            timeout=timeout_sec,
+            initializer=initializer,
+        )
+        self.timeout_sec = timeout_sec
+        self.f_build = f_build
+        self.f_export = f_export
+        self._sanity_check()
+
+    def build(self, build_inputs: List[BuilderInput]) -> List[BuilderResult]:
+        results: List[BuilderResult] = []
+        map_result: MapResult
+
+        # Dispatch the build inputs to the worker processes.
+        for map_result in self.pool.map_with_error_catching(
+            lambda x: LocalBuilder._worker_func(*x),
+            [
+                (
+                    self.f_build,
+                    self.f_export,
+                    build_input.mod,
+                    build_input.target,
+                )
+                for build_input in build_inputs
+            ],
+        ):
+            if map_result.status == StatusKind.COMPLETE:
+                results.append(BuilderResult(map_result.value, None))
+            elif map_result.status == StatusKind.TIMEOUT:
+                results.append(
+                    BuilderResult(
+                        None,
+                        f"LocalBuilder: Timeout, killed after {self.timeout_sec} seconds",
+                    )
+                )
+            elif map_result.status == StatusKind.EXCEPTION:
+                results.append(
+                    BuilderResult(
+                        None,
+                        "LocalBuilder: An exception occurred\n" + str(map_result.value),
+                    )
+                )
+            else:
+                raise NotImplementedError(map_result.status)
```

Comment:
Should we raise an exception or just return the not implemented status to keep it going?

Reply:
This branch is actually unreachable, given the StatusKind only has 4 possible outcomes:
- RUNNING => it won't appear in our case, because popen pool enforces join semantics
- COMPLETE => handled in the 1st branch
- EXCEPTION => handled in the 3rd branch
- TIMEOUT => handled in the 2nd branch

So actually it doesn't matter whatever we raise, because the branch is unreachable. What do you think is a good default for this case?

Reply:
Then I think we could just raise an exception with a message saying it is unreachable. The purpose is to let developers know we don't need to worry about that.

Reply:
Got it. Just updated the PR with a more detailed description indicating that the branch is unreachable